src/libaktualizr/package_manager/ostreemanager.cc

```diff
@@ -234,12 +234,23 @@ OstreeManager::OstreeManager(PackageConfig pconfig, std::shared_ptr<INvStorage>
 bool OstreeManager::fetchTarget(const Uptane::Target &target, Uptane::Fetcher &fetcher, const KeyManager &keys,
                                 FetcherProgressCb progress_cb, const api::FlowControlToken *token) {
   if (!target.IsOstree()) {
+    // The case when the ostree package manager is set as a package manager for aktualizr
+    // while the target is aimed for a secondary ECU that is configured with another/non-ostree package manager
     return PackageManagerInterface::fetchTarget(target, fetcher, keys, progress_cb, token);
   }
   return OstreeManager::pull(config.sysroot, config.ostree_server, keys, target, token, progress_cb).success;
 }
 
 TargetStatus OstreeManager::verifyTarget(const Uptane::Target &target) const {
+  if (!target.IsOstree()) {
```

Comment:
I think we may need a similar check in dockerappmanager.cc? In fact I think the current logic isn't quite right there. Its doing verification for all target types and really should just run when the type is OSTree. I think something like this is probably correct:
~~~
TargetStatus DockerAppManager::verifyTarget(const Uptane::Target &target) const {
  TargetStatus status = OstreeManager::verifyTarget(target);
  if (status != TargetStatus::kGood || !target.IsOstree()) {
      return status;
  }
  auto cb = [this](const std::string &app, const Uptane::Target &app_target) {
    LOG_INFO << "Verifying " << app << " -> " << app_target;
    std::stringstream ss;
    ss << *storage_->openTargetFile(app_target);
    DockerApp dapp(app, config);
    return dapp.render(ss.str(), false);
  };
  if (!iterate_apps(target, cb)) {
    return TargetStatus::kInvalid;
  }
  return TargetStatus::kGood;
}
~~~

Reply:
If secondaries/multi-ECU update is not required in case of "ostree+docker-app" package manager usage on Primary then the current logic is fine provided that it works now for updating both ostree and docker app on Primary.

My understanding is that the "ostree+docker-app" package manager is intended for supporting of an ostree-based rootfs update and docker apps configuration update on Primary ? If it's so I am not sure if I understand why `auto res = OstreeManager::install(target);` is called in `DockerAppManager::install()`, I suppose, the logic should call the ostree install routine only if the target type is ostree?

Reply:
What value of `target.type_` will be in case of the docker-app update ?

Reply:
We are planning to redesign approach to package management as new requirements/use cases appeared:

1) secondaries support
2) multiple type of update on a single ECU/Primary
3) providing an integrator with  possibility  to add a new type of 'package manager' without libaktualizr modification.

Reply:
@mike-sul - The docker-app manager is confusing right now because I haven't documented it well at all. I think what I should do is put a PR that includes some code comments about the package manager so people can reason about it. Then I can fix it and add some tests to make sure its following the logic you've done here properly.

Reply:
FYI - here's the blog https://foundries.io/insights/2019/09/23/docker-app/

Reply:
> FYI - here's the blog https://foundries.io/insights/2019/09/23/docker-app/

Thanks for pointing out to it, gonna read it through.